Attitude Quaternion Stream with invalid quaternion

Since PR #12596 , the Firmware sends the mavlink message ATTITUDE_QUATERNION with an invalid (0, 0, 0, 0) quaternion until the ekf2 starts to publish the `vehicle_attitude`. 
The mavlink message is published bacause the condition is that either `vehicle_angular_velocity`  or `vehicle_attitude` is updated and `vehicle_angular_velocity`  is always ready before the attitude. 

https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_messages.cpp#L1289

For ROS user, this leads to a bunch of error messages on start up
``` bash 
[ERROR] [1569826119.454300726, 0.856000000]: Ignoring transform for child_frame_id "fcu" from authority "unknown_publisher" because of an invalid quaternion in the transform (0.000000 0.000000 0.000000 0.000000)
```

I have tried to publish an invalid quaternion (NAN, NAN, NAN, NAN) but I get the same error message because ROS requires the quaternion to be valid and this value is accessed multiple times across multiple MAVROS plugins. 

I assume the OR condition in the update decision was a conscious choice. How can we prevent to send out invalid quaternions?

--- a/src/modules/mavlink/mavlink_messages.cpp
+++ b/src/modules/mavlink/mavlink_messages.cpp
@@ -1185,7 +1185,6 @@ private:
 	MavlinkOrbSubscription *_att_sub;
 	MavlinkOrbSubscription *_angular_velocity_sub;
 	uint64_t _att_time{0};
-	uint64_t _angular_velocity_time{0};
 
 	/* do not allow top copying this class */
 	MavlinkStreamAttitude(MavlinkStreamAttitude &) = delete;
@@ -1200,18 +1199,16 @@ protected:
 
 	bool send(const hrt_abstime t)
 	{
-		bool updated = false;
+		vehicle_attitude_s att;
 
-		vehicle_attitude_s att{};
-		vehicle_angular_velocity_s angular_velocity{};
-		updated |= _att_sub->update(&_att_time, &att);
-		updated |= _angular_velocity_sub->update(&_angular_velocity_time, &angular_velocity);
+		if (_att_sub->update(&_att_time, &att)) {
+			vehicle_angular_velocity_s angular_velocity{};
+			_angular_velocity_sub->update(&angular_velocity);
 
-		if (updated) {
 			mavlink_attitude_t msg{};
 
 			const matrix::Eulerf euler = matrix::Quatf(att.q);
-			msg.time_boot_ms = math::max(angular_velocity.timestamp, att.timestamp) / 1000;
+			msg.time_boot_ms = att.timestamp / 1000;
 			msg.roll = euler.phi();
 			msg.pitch = euler.theta();
 			msg.yaw = euler.psi();
@@ -1267,7 +1264,6 @@ private:
 	MavlinkOrbSubscription *_att_sub;
 	MavlinkOrbSubscription *_angular_velocity_sub;
 	uint64_t _att_time{0};
-	uint64_t _angular_velocity_time{0};
 
 	/* do not allow top copying this class */
 	MavlinkStreamAttitudeQuaternion(MavlinkStreamAttitudeQuaternion &) = delete;
@@ -1281,17 +1277,15 @@ protected:
 
 	bool send(const hrt_abstime t)
 	{
-		bool updated = false;
+		vehicle_attitude_s att;
 
-		vehicle_attitude_s att{};
-		vehicle_angular_velocity_s angular_velocity{};
-		updated |= _att_sub->update(&_att_time, &att);
-		updated |= _angular_velocity_sub->update(&_angular_velocity_time, &angular_velocity);
+		if (_att_sub->update(&_att_time, &att)) {
+			vehicle_angular_velocity_s angular_velocity{};
+			_angular_velocity_sub->update(&angular_velocity);
 
-		if (updated) {
 			mavlink_attitude_quaternion_t msg{};
 
-			msg.time_boot_ms = math::max(angular_velocity.timestamp, att.timestamp) / 1000;
+			msg.time_boot_ms = att.timestamp / 1000;
 			msg.q1 = att.q[0];
 			msg.q2 = att.q[1];
 			msg.q3 = att.q[2];

